### ISSUE TITLE
Push notifications sometimes disappear immediately

### ISSUE THREAD

**Issue by johns2s** [vector-im/element-android#3996]
In a 1:1 encrypted chat, I've noticed that often (but not always, maybe 2/3 of the time), notifications disappear shortly after they appear and my phone vibrates. When this happens, the dot signifying new notifications on the element app icon also disappears.

I'm currently using version 1.2.1 on a Pixel 5 running Android 11, but I've noticed this issue for a few months on both this device and a Pixel 3, also running Android 11.

[Here's a video](https://streamable.com/focnm4)

**Comment by saintger** [comment]
I can confirm that I have the same issue.
Previously I had this issue when element was left running in the background on my Mac desktop, but now I also have the bug with no other running instances.
I am running Element version 1.2.0 (no new version available as of today on my phone) on OnePlus 7 Pro with Android 11 (Oxygen OS 11.0.2.1.GM21BA).

**Comment by johns2s** [comment]
I've also experienced this in FluffyChat. Upon further investigation it looks like for some reason these messages are sometimes marked as read erroneously, causing the notifications to disappear.

**Comment by vlladdrakk** [comment]
My wife has this issue on her pixel 2 running android 11. Any time I send her a message I see that she has immediately read it, but she hasn't and the notification disappears.

**Comment by bmarty** [comment]
It can happen if another client is viewing the room and moves the read receipt, but I guess it is not the case here?

**Comment by ouchadam** [comment]
it would be quite strange for the android app to move the read receipt on receiving the push notification, although the logic does exist on the _mark read_ notification action...

to double check does this _mark as read_ behaviour happen when the android device is off or in airplane mode? this should pinpoint if there's an app problem or a potential client running in the background somewhere else

**Comment by lindhe** [comment]
I've also got this issue. Also from a 1:1 encrypted room.

**Comment by saintger** [comment]
I have updated to version 1.3.7 and I don't have the issue anymore.
If the issue comes again, I will report it here (and try with the device off !)

**Comment by lindhe** [comment]
I have updated to 1.3.7, and now the notification will not disappear after I click it or choose "mark as read" at all! 😅 I have to manually dismiss it by swiping it to the side.

**Comment by ouchadam** [comment]
@lindhe that's unexpected :thinking: would you be able to create a new issue? Would be great to have some details about your device/OS 

There were some changes needed for the upgrade to android 12 specifically around notification click https://github.com/vector-im/element-android/pull/4433 If you're running android 12 these might fix it

**Comment by lindhe** [comment]
Yeah, I'll create a new issue if I encounter it again. Just thought it was funny if disappearing notifications was fixed by making them permanent. ;) But maybe it was just a fluke, so I'll wait out another one before creating anything.

I'm running Android 11, but expect to upgrade to 12 within a few weeks.

**Comment by ouchadam** [comment]
fix one issue and create two more :joy: if you are able to reproduce and create a ticket, I'll jump onto it straight away

**Comment by ouchadam** [comment]
I've been able to reproduce the non dismissing notification issue, have a fix on the way https://github.com/vector-im/element-android/issues/4505

**Comment by saintger** [comment]
Unfortunately it seems that I got the issue again.
I am running Element version 1.3.7 (no new version available as of today on my phone) on OnePlus 7 Pro with Android 11 (Oxygen OS 11.0.2.1.GM21BA).
As soon as I send the message, I can see that the message was immediately automatically mark as read even though we didn't read the message.
I can confirm that this is related to Element on Android, because when I switch the phone in airplane mode, the message is not automatically read.

**Comment by saintger** [comment]
Hello,
With version 1.3.15, it is still not working.
I don't receive any notification, but this time the message is not marked as read (which can be considered as an improvement).
I've tried to use the menu entry to debug the notification and all 9 steps executed correctly without any issues.

**Comment by ouchadam** [comment]
@saintger this sounds like you're experiencing #5038 (which has a fix on the way!)

**Comment by saintger** [comment]
@ouchadam the issue was in the comments in your link: it is due to Adguard which is blocking firebaseinstallations.googleapis.com !
It is strange that it wasn't detected by the menu entry to debug the notification (the 9 steps) but anyway I manually put this domain the whitelist and now notifications seem to work again.
I'll monitor how i goes in the following days/weeks and see if the bug here is happening again.
Thanks for the hint !

**Comment by svenb1234** [comment]
This issue appeared on our devices some weeks ago. The only other "client" we have are Smartwatches which show the notification. However it worked before with the watches.

It seems to happen more often if the screen is on because one is using another app.

**Comment by svenb1234** [comment]
I was told that a fix has already been merged and will be part of the next release. No idea though when that is.

**Comment by thomaspeissl** [comment]
Please release the latest app version on Google Play store.

**Comment by svenb1234** [comment]
Yep, #7789 seems stalled. Maybe Google did not reply/release yet.

**Comment by SPiRiT369** [comment]
As I commented in #7232, I found that if you press Back in the app to return to the chats list, then notifications will work fine. At least in my Samsung phone it helps. Maybe if you are inside a room/DM, the app thinks you are reading the new messages even though the app is minimized. That's my guess :/

**Comment by thomaspeissl** [comment]
With the new released Element App Version 1.5.20 from Google Play Store this issue seems fixed for me.

**Comment by TLATER** [comment]
I am still experiencing this with 1.5.28 from the Google Play Store. I ran into this previously shortly before it was fixed, but it resolved after a full reinstall. It has now reappeared (and been problematic again for a week or two).

I'm not getting any notifications from a 1:1 encrypted chat (don't have other 1:1 chats), but *am* getting notifications as expected from unencrypted rooms with more users, including bots.

I have a pebble smartwatch, and it *does* vibrate whenever I get a message, however the notifications are dismissed almost instantly - my phone does not make a sound or vibrate, but occasionally the notification dot does appear for a split second. All notifications are logged in the notification history.

I think it's the same symptoms as people are reporting here?

**Comment by Xinayder** [comment]
I get a similar issue as described by @TLATER. I don't know what happened to cause the issue, but I am using a bridged room with WhatsApp using the Mautrix bridge and after about a month running the bridge, the notifications for a specific group in WhatsApp stopped working. I receive a message, my phone vibrates and Element pops up but is quickly dismissed on the status bar. The only way I find out that I received a message for that specific group is that WhatsApp shows up in the notification list (but vibration is muted for it). I also have problems marking messages as read in that group inside Element, it keeps increasing the amount of unread notifications even though I've read them.

**Comment by TLATER** [comment]
Just to contribute my two cents here: I, again, now get notifications after clearing storage/cache and logging back in. I'll make sure to come back if it breaks again; happy to provide any logs that could help identify the issue.

I don't like clearing my login because of vectori-im/element-web#16413 and related issues that make encryption a PITA whenever I do so, but at least I don't feel like I need to have my smartwatch on 100% of the time.

**Comment by TLATER** [comment]
It's still happening periodically, sadly. My workaround at this point is just to clear data whenever I notice I missed a notification and re-log.

**Comment by tommie** [comment]
@TLATER have you had any changes with this in the three years that have passed? I recently started encountering this after switching phones.

But I'm on a self-hosted Dendrite server, and I'm wondering if you are on matrix.org, or self-hosted, to rule out the server side.

**Comment by TLATER** [comment]
I have switched to Element X, and it's been a while since I've noticed missing a notification. I self-host a continuwuity instance (conduit 3 years ago).

That said, I've also stopped using Matrix quite as actively, life happened and for better or worse my social circles have shifted, so I get a lot fewer notifications.

**Comment by tommie** [comment]
That's one way... :)

I've been contemplating switching away, since it turned out Dendrite is a dead end. But not yet.
